osu.Game/Beatmaps/WorkingBeatmap.cs

```diff
@@ -353,4 +355,186 @@ public BeatmapLoadTimeoutException(BeatmapInfo beatmapInfo)
             }
         }
     }
+
+    /// <summary>
+    /// Temporary helper extension methods to aid in using <see cref="LoggingTrack"/> with minimal code damage.
+    /// </summary>
+    internal static class TrackExtensions
+    {
+        public static Track GetUnderlyingTrack(this ITrack track)
+        {
+            if (track is Track t)
+                return t;
+
+            if (track is LoggingTrack lt)
+                return lt.UnderlyingTrack;
+
+            return null;
+        }
+
+        public static bool IsLoaded(this ITrack track)
+        {
+            if (track is Track t)
+                return t.IsLoaded;
+
+            if (track is LoggingTrack lt)
+                return lt.UnderlyingTrack.IsLoaded;
+
+            return false;
+        }
```

Comment:
Is this required? `LoggingTrack` implements this as a redirect to `UnderlyingTrack.IsLoaded`.